projects/observability/src/shared/components/span-detail/span-detail.component.ts

```diff
@@ -79,12 +99,39 @@ export class SpanDetailComponent implements OnChanges {
   @Output()
   public readonly closed: EventEmitter<void> = new EventEmitter<void>();
 
+  @ViewChild('requestTemplate', { static: true })
+  public requestTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('responseTemplate', { static: true })
+  public responseTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('attributesTemplate', { static: true })
+  public attributesTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('exitCallsTemplate', { static: true })
+  public exitCallsTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('logEventsTemplate', { static: true })
+  public logEventsTemplate!: TemplateRef<unknown>;
+
   public showRequestTab?: boolean;
   public showResponseTab?: boolean;
   public showExitCallsTab?: boolean;
   public showLogEventsTab?: boolean;
   public totalLogEvents?: number;
 
+  public tabs: ToggleItem<SpanDetailTab>[] = [];
+
+  private readonly activeTabSubject: BehaviorSubject<ToggleItem<SpanDetailTab> | undefined> = new BehaviorSubject<
```

Comment:
do we really need this? can we not do it with a simple variable?

Reply:
Since we rely on observable for everything and this event can trigger side effects, thought a subject will make more sense here.